controllers/ctltools.js

```diff
@@ -27,6 +27,7 @@ const sanitizeToolInputs = (id, body) => {
     body.openNewTab = !!body.openNewTab;
     body.isHidden = !!body.isHidden;
     body.config.baseUrl = body.config.baseUrl || undefined;
+    body.restrictToContexts = Array.isArray(body.restrictToContexts) ? body.restrictToContexts : [].concat(body.restrictToContexts || []);
```

Comment:
is there the possibility for restrictToContexts to not be an array? you can use `body.restrictToContexts ?? []` for null checks, if its only that.

Reply:
you will receive this from the input field, in case it is:
empty -> [ undefined ] <- array with one element of undefined
one context -> "course" <- as a String
two contexts -> ["course", "board-element"]

Reply:
but then you have to transform the "empty"-case to an empty array, but currently you return `[undefined]`